vulnerabilities/pipelines/v2_importers/collect_repo_fix_commits.py

```diff
@@ -0,0 +1,117 @@
+import os
+import re
+import shutil
+import subprocess
+import tempfile
+from collections import defaultdict
+
+from git import Repo
+
+from vulnerabilities.importer import AdvisoryData
+from vulnerabilities.importer import ReferenceV2
+from vulnerabilities.pipelines import VulnerableCodeBaseImporterPipelineV2
+
+SECURITY_PATTERNS = [
+    r"\bCVE-\d{4}-\d{4,19}\b",
+    r"\bGHSA-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}\b",
+    r"\bPYSEC-\d{4}-\d{1,6}\b",
+    r"\bXSA-\d{1,4}\b",
+]
+
+
+class CollectRepoFixCommitPipeline(VulnerableCodeBaseImporterPipelineV2):
+    """
+    Pipeline to collect fix commits from any git repository.
+    """
+
+    pipeline_id = "repo_fix_commit"
+
+    @classmethod
+    def steps(cls):
+        return (
+            cls.clone,
+            cls.collect_and_store_advisories,
+            cls.clean_downloads,
+        )
+
+    def clone(self):
+        """Clone the repository."""
+        self.repo_url = "https://github.com/torvalds/linux"
+        repo_path = tempfile.mkdtemp()
+        cmd = [
+            "git",
+            "clone",
+            "--bare",
+            "--filter=blob:none",
+            "--no-checkout",
+            self.repo_url,
+            repo_path,
+        ]
+        subprocess.run(cmd, check=True)
+        self.repo = Repo(repo_path)
```

Comment:
Since we already using gitpython let's use the same for cloning too.

```suggestion
        self.repo = Repo.clone_from(
                url=self.repo_url,
                to_path=repo_path,
                bare=True,
                no_checkout=True,
                multi_options=["--filter=blob:none"]
        )
```